projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -525,6 +535,57 @@ function wpcom_track_publish_first_post_task() {
 	wpcom_mark_launchpad_task_complete_if_active( 'first_post_published_newsletter' );
 }
 
+/**
+ * Gets the number of published posts that were not created by Headstart.
+ *
+ * @return int
+ */
+function wpcom_get_published_non_headstart_posts_count() {
+	// Ensure that Headstart posts don't affect the count
+	if ( defined( 'HEADSTART' ) && HEADSTART ) {
+		return 0;
+	}
+
+	// Make sure we don't count any published posts that were created by Headstart.
+	$headstart_posts_filter = array(
+		'post_type'    => 'post',
+		'fields'       => 'ids',
+		'status'       => 'publish',
+		'meta_key'     => '_hs_old_id',
+		'meta_compare' => 'EXISTS',
+	);
+
+	$total_posts_count     = wp_count_posts( 'post' )->publish;
+	$headstart_posts_count = count( get_posts( $headstart_posts_filter ) );
+
+	return $total_posts_count - $headstart_posts_count;
+}
+
+/**
+ * Callback for completing the Write 3 posts task.
+ *
+ * @return void
+ */
+function wpcom_track_write_3_posts_task() {
+	$published_non_headstart_posts = wpcom_get_published_non_headstart_posts_count();
+
+	if ( $published_non_headstart_posts >= 3 ) {
+		wpcom_mark_launchpad_task_complete_if_active( 'write_3_posts' );
+	}
+}
+
+/**
+ * Callback for getting the number of posts published.
+ *
+ * @param array $task The Task definition.
+ * @return int
+ */
+function wpcom_get_write_3_posts_repetition_count( $task ) {
+	$published_non_headstart_posts = wpcom_get_published_non_headstart_posts_count();
+
+	return min( $task['target_repetitions'], $published_non_headstart_posts );
+}
```

Comment:
General nit: I think we should be prefixing all these functions with `wpcom_launchpad_`, not just `wpcom_`.
```suggestion
function wpcom_launchpad_get_published_non_headstart_posts_count() {
	// Ensure that Headstart posts don't affect the count
	if ( defined( 'HEADSTART' ) && HEADSTART ) {
		return 0;
	}

	// Make sure we don't count any published posts that were created by Headstart.
	$headstart_posts_filter = array(
		'post_type'    => 'post',
		'fields'       => 'ids',
		'status'       => 'publish',
		'meta_key'     => '_hs_old_id',
		'meta_compare' => 'EXISTS',
	);

	$total_posts_count     = wp_count_posts( 'post' )->publish;
	$headstart_posts_count = count( get_posts( $headstart_posts_filter ) );

	return $total_posts_count - $headstart_posts_count;
}

/**
 * Callback for completing the Write 3 posts task.
 *
 * @return void
 */
function wpcom_launchpad_track_write_3_posts_task() {
	$published_non_headstart_posts = wpcom_launchpad_get_published_non_headstart_posts_count();

	if ( $published_non_headstart_posts >= 3 ) {
		wpcom_mark_launchpad_task_complete_if_active( 'write_3_posts' );
	}
}

/**
 * Callback for getting the number of posts published.
 *
 * @param array $task The Task definition.
 * @return int
 */
function wpcom_launchpad_get_write_3_posts_repetition_count( $task ) {
	$published_non_headstart_posts = wpcom_launchpad_get_published_non_headstart_posts_count();

	return min( $task['target_repetitions'], $published_non_headstart_posts );
}
```